docs/design/phases.md

```diff
@@ -0,0 +1,110 @@
+## Phases in kubeadm
+
+`kubeadm init` does a lot of things. In order to make kubeadm the common building block for Kubernetes we want it to be, each logical phase kubeadm does during `init` should be invokable separately. Here's an explanation what kubeadm does in the different phases:
```

Comment:
Suggested edit:

`kubeadm init` does a lot. In order to make kubeadm the common building block for Kubernetes, each logical phase kubeadm does during `init` should be invokable separately. Here's an explanation what kubeadm does in these different phases: